docs/_releases/v2.0.0-pre.4/matchers.md

```diff
@@ -112,6 +112,26 @@ Requires the value to be a `Function`.
 Requires the value to be an `Array`.
 
 
+#### `sinon.match.array.deepEquals(iterable)`
+
+Requires an iterable value to be deep equal another one.
```

Comment:
@fearphage I agree, I also raised the same question in [a comment](https://github.com/sinonjs/sinon/pull/1208#discussion_r93198894) at my PR at #1208.

> However, I was just thinking if we should act upon those iterables as if they were arrays, after all this matcher has the word array in it.

Since we're calling methods from `Array.prototype` and passing a value to be used as `this` it would work if users had their own array implementations or with things like `arguments`.

I can change this if you want me too, I'd like to do whatever you prefer. If we want to be more strict it would be good to accept `Arrays` only, but if we want to allow other similar implementations this is the way to go.

Maybe we could also have an `match.iterable` or something like that.

What do you think it's better? Let me know and we can fix this if needed 😄